### ISSUE TITLE
Auto Update cannot be deferred and cannot be configured

### ISSUE THREAD

**Issue by andrewsensus** [docker/for-win#10139]
- [x ] I have tried with the latest version of Docker Desktop
  - N/A: [ ] I have tried disabling enabled experimental features
  - [ ] I have uploaded Diagnostics
  - Diagnostics ID:
  - [ ] I am using the Docker WSL2 Backend
  - [x ] I am using the Docker Hyper-V Backend

### Actual behavior
Docker updates automatically, even on just a Windows restart.
There is no configuration to do otherwise in the General tab of Settings.

### Expected behavior
Docker should not update without giving the user the ability to proceed or defer the update. The settings should include an option to disable automatic checks for updates.

### Information
This is reproducible. Starting with about 3.x, there has not been configuration in the expected location and updates have happened without prompting me to confirm or defer them.
Prior to that time I was able to accept updates or defer them, and I did so as sometimes updates broke my Docker Desktop install, and I want control of when I take that risk.

  - Windows Version: Windows 10, Version 1909, Build 18363.1256
  - Docker Desktop Version: the last several this morning it automatically updated from 3.0.(4?) to 3.1.0 when I restarted Windows without giving me a choice
  - WSL2 or Hyper-V backend?  Hyper-V
  - Are you running inside a virtualized Windows e.g. on a cloud server or on a mac VM: no

### Steps to reproduce the behavior
1. Observe the General tab of the settings Dialog

Also

1. Wait for a new version of Docker Desktop to come out
2. Restart Windows

**Comment by stephen-turner** [comment]
This is by design, but see docker/roadmap#183 for the request to change it. I'll close this ticket as a dupe of that one.

**Comment by andrewsensus** [comment]
See also https://github.com/docker/for-mac/issues/5119

**Comment by patricknelson** [comment]
That is VERY unfortunate 😞.  For example, the auto-updated fails on my machine. I found that I have been able to manually upgrade (usually by manually uninstalling and then installing the new version). 

As a result, without any workaround, Docker is constantly broken with an exclamation point "!" on the icon notifying me of an update that is not able to install. The only workaround that I've found to manually hack Docker for Windows is to explicitly deny write permissions so it cannot download installers.  Basically:

1. Navigate to `%LOCALAPPDATA%\Docker Desktop Installer%` directory (e.g. `C:\Users\USERNAME\AppData\Local\Docker Desktop Installer`)
2. Delete the existing installer `exe` that may already be present (if applicable)
3. Right click on that folder, either in the navigation pane or somewhere in the whitespace and select "Properties"
4. Go to the "Security" tab
5. Click "Edit" to change permissions 
6. Uncheck "Modify" permissions and click "Ok" on all dialogs 

**This really shouldn't be necessary.** 🤨
